### PR TITLE
feat(bigtable): adds checkandmutate to proxy

### PR DESCRIPTION
**Pull request by telpirion**

### CODE REVIEW

File: bigtable/internal/testproxy/proxy.go [hunk 1]
```diff
@@ -789,7 +789,58 @@ func (s *goTestProxyServer) BulkMutateRows(ctx context.Context, req *pb.MutateRo
 // CheckAndMutateRow responds to the CheckAndMutateRow RPC. This method applies
 // one mutation if a condition is true and another mutation if it is false.
 func (s *goTestProxyServer) CheckAndMutateRow(ctx context.Context, req *pb.CheckAndMutateRowRequest) (*pb.CheckAndMutateRowResult, error) {
-	return nil, stat.Error(codes.Unimplemented, "CheckAndMutateRow not implemented")
+	s.clientsLock.Lock()
```
Comment (enocom) [inline bigtable/internal/testproxy/proxy.go:792]: RLock() and RUnlock() here.

Comment (enocom) [reply bigtable/internal/testproxy/proxy.go:792]: (Also it seems we've been missing this in the other methods)

File: bigtable/internal/testproxy/proxy.go [hunk 2]
```diff
@@ -789,7 +789,58 @@ func (s *goTestProxyServer) BulkMutateRows(ctx context.Context, req *pb.MutateRo
...
+	f := bigtable.PassAllFilter()
+
+	if rfPb != nil {
+		f = *filterFromProto(rfPb)
+	}
+
+	c := bigtable.NewCondMutation(f, trueMuts, falseMuts)
+
+	var matched bool
+	ao := bigtable.GetCondMutationResult(&matched)
```
Comment (enocom) [inline bigtable/internal/testproxy/proxy.go:821]: nit: let's keep declarations closest to their use. This `ao` and matched could move down to line 823

**Review by enocom — CHANGES_REQUESTED** [review]

File: bigtable/internal/testproxy/proxy.go (hunk 1 above)
Comment (telpirion) [reply bigtable/internal/testproxy/proxy.go]: Done.

File: bigtable/internal/testproxy/proxy.go (hunk 2 above)
Comment (telpirion) [reply bigtable/internal/testproxy/proxy.go]: Done.

**Review by enocom — APPROVED** [review]

**Pull request merged**